hack/update-codegen.sh

```diff
@@ -26,8 +26,12 @@ go mod vendor
 
 CODEGEN_PKG=${CODEGEN_PKG:-$(ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
 
+# `--output-base $SCRIPT_ROOT` will output generated code to current dir
 bash "${CODEGEN_PKG}"/generate-groups.sh "deepcopy,client,informer,lister" \
     github.com/pingcap/tidb-operator/pkg/client \
     github.com/pingcap/tidb-operator/pkg/apis \
     pingcap:v1alpha1 \
+    --output-base $SCRIPT_ROOT \
     --go-header-file ./hack/boilerplate/boilerplate.generatego.txt
+# then we merge generated code with our code base and clean up
+cp -r github.com/pingcap/tidb-operator/pkg $SCRIPT_ROOT && rm -rf github.com
```

Comment:
Does this work for both the cases where the code base is under $GOPATH or not?

Reply:
Could you please also help update the `CONTRIBUTING.md` to make it clear that if contributors update the CRD related types definition, they should run the `hack/update-all.sh`?

Reply:
https://github.com/pingcap/tidb-operator/pull/3519#discussion_r530263962